myia/impl_py.py

```diff
@@ -0,0 +1,160 @@
+"""Implementations for the debug VM."""
+
+
+from typing import Dict, Callable
+from copy import copy
```

Comment:
If you really want to enforce immutability, this might have to be `deepcopy` instead of `copy`?

Reply:
This is for the implementation of `setitem` and `setattr`, which only modify the object shallowly.

Reply:
Well, that depends on the object. Which ones are you trying to support here? `setitem` is for lists and arrays I guess, but what is `setattr` for?

Reply:
`setattr` would be for records, but I suppose that may be a bit premature.